apps/web-app/components/settings/addNewGroup.vue

```diff
@@ -4,47 +4,264 @@
+    description: String | null;
+    color: String | null;
+  }
+  const selectedOrgGroups = ref<OrgUserGroups[]>([]);
+
+  watch(orgDomainsData, (newOrgDomainsData) => {
+    if (newOrgDomainsData?.domainData) {
+      for (const domain of newOrgDomainsData.domainData) {
+        orgDomains.value.push({
+          domainPublicId: domain.publicId,
+          domain: domain.domain
+        });
+      }
+    }
+  });
 
   const formValid = computed(() => {
+    if (createEmailIdentityForGroup.value) {
+      if (isPro.value === false) {
+        return (
+          newIdentityUsernameValid.value === true &&
+          (newIdentitySendNameValid.value || newGroupNameValid.value) ===
+            true &&
+          selectedDomain.value?.domainPublicId &&
+          newGroupNameValid.value === true &&
+          !!newGroupColorValue.value
+        );
+      }
+      return (
+        newIdentityUsernameValid.value === true &&
+        (newIdentitySendNameValid.value || newGroupNameValid.value) === true &&
+        selectedDomain.value?.domainPublicId &&
+        !!newGroupColorValue.value
+      );
+    }
     return newGroupNameValid.value === true && !!newGroupColorValue.value;
   });
 
+  const toggleableOptions = ref<{
+    address: boolean;
+    sendName: boolean;
+    domainName: boolean;
+  }>({
+    address: false,
+    sendName: false,
+    domainName: false
+  });
+
+  const toggleInputOption = (option: keyof typeof toggleableOptions.value) => {
+    toggleableOptions.value[option] = !toggleableOptions.value[option];
+    if (option === 'address' && !toggleableOptions.value.address) {
+      newIdentityUsernameValid.value = null;
+      newIdentityUsernameValidationMessage.value = '';
+    }
+    if (option === 'sendName' && !toggleableOptions.value.sendName) {
+      newIdentitySendNameValid.value = null;
+    }
+    if (option === 'domainName' && !toggleableOptions.value.domainName) {
+      newDomainNameValid.value = null;
+    }
+  };
+
   const emit = defineEmits(['close']);
 
   const route = useRoute();
 
   const orgSlug = route.params.orgSlug as string;
 
   async function createGroup() {
-    if (!newGroupColorValue.value) return;
-    buttonLoading.value = true;
-    buttonLabel.value = 'Creating...';
-    const { newGroupPublicId } =
-      await $trpc.org.users.userGroups.createOrgUserGroups.mutate({
-        groupName: newGroupNameValue.value,
-        groupDescription: newGroupDescriptionValue.value,
-        groupColor: newGroupColorValue.value
+    if (createEmailIdentityForGroup.value) {
+      if (!newGroupColorValue.value) return;
+      buttonLoading.value = true;
+      buttonLabel.value = 'Creating...';
+      const toast = useToast();
+      const { newGroupPublicId } =
+        await $trpc.org.users.userGroups.createOrgUserGroups.mutate({
+          groupName: newGroupNameValue.value,
+          groupDescription: newGroupDescriptionValue.value,
+          groupColor: newGroupColorValue.value
+        });
```

Comment:
this code here is duplicated
consider moving it out of the if loop to just before the if check
we can then remove the duplication from the else statement